bitnami/discourse/templates/NOTES.txt

```diff
@@ -133,10 +150,18 @@ This deployment will be incomplete until you configure Discourse with a resolvab
     --set global.imagePullSecrets={{ .Values.global.imagePullSecrets }} \
     {{- end }}{{- end }}
     {{- if (and .Values.redis.enabled .Values.redis.usePassword) }}
+    {{- if .Values.redis.existingSecret }}
+    --set redis.existingSecret={{ include "discourse.redis.secretName" . }} \
+    {{- else if not (empty .Values.redis.password) }}
     --set redis.password=$APP_REDIS_PASSWORD \
-    {{- else if and (not .Values.redis.enabled) (not (empty .Values.externalRedis.password)) }}
-    --set redis.externalRedis=$APP_REDIS_PASSWORD \
+    {{- end }}
+    {{- else if (not .Values.redis.enabled) }}
     --set redis.enabled=false \
+    {{- if .Values.externalRedis.existingSecret }}
+    --set externalRedis.existingSecret={{ include "discourse.redis.secretName" . }} \
```

Comment:
This is great but not necessary. The reason behind setting specific values when performing an upgrade is because there are certain fields (such as `passwords`) that are autogenerated if the final user does not provide with one. 

Imagine this scenario:

**1st Release** 
redis.password='' ==> The Chart autogenerates a random password, let's call it `alpha`

**2nd Release** 
We perform an upgrade from our previous release. Remember that `redis.password='' ` ==> The Chart will misunderstand this and replace `alpha` by a new random password, let's call it `beta`

Discourse cannot connect to Redis because it thinks `beta` is now the password!

However, if we explicitly set these parameter to fixed values, the release will be performed correctly 😁.